File: client/src/app/models/project.ts
export class Project {

    id: number;
    name: string;
    deadline: Date;
    complete: boolean;
    timeRegistrations: [];

    constructor(id: number, name: string, deadline: Date, complete: boolean, timeRegistrations: []) {
        this.id = id;
        this.name = name;
        this.deadline = deadline;
        this.complete = complete;
        this.timeRegistrations = timeRegistrations;
    }
  }

File: server/Timelogger/Entities/Project.cs
using System;
using System.Collections.Generic;

namespace Timelogger.Entities
{
	public class Project
	{
		public int Id { get; set; }
		public string Name { get; set; }
		public DateTime Deadline { get; set; }
		public bool Complete { get; set; } = false;

		public List<TimeRegistration> TimeRegistrations { get; set; } = new List<TimeRegistration>();
	}
}
